src/StrawberryfieldHydroponicsService.php

```diff
@@ -181,64 +297,116 @@ public function run() {
 
         $pid = exec(
           sprintf("%s > /dev/null 2>&1 & echo $!", $cmd)
+          //sprintf("%s > /dev/null & echo $!", $cmd)
         );
         \Drupal::state()->set('hydroponics.queurunner_last_pid', $pid);
         $this->logger->info('PID for Hydroponics Service: @pid', [
             '@pid' => $pid]
         );
-      } else {
+      }
+      //The normal running condition when all is ok
+      elseif (($deltaTime <= $heartbeat_max_delta) && ($queuerunner_pid > 0) && $running_posix) {
         $this->logger->info('Hydroponics Service already running with PID @pid, time passed since last seen @time', [
-            '@time' => ($currentTime - $lastRunTime),
-            '@pid' => $queuerunner_pid
+          '@time' => $deltaTime,
+          '@pid' => $queuerunner_pid
+          ]
+        );
+      }
+      //Something went wrong so log error and no action
+      else {
+        $this->logger->error('Hydroponics Service error status: PID @pid, time passed since last seen @time, on Process table @ptable', [
+          '@time' => $deltaTime,
+          '@pid' => $queuerunner_pid,
+          '@ptable' => ($running_posix) ? "YES" : "NO"
           ]
         );
       }
     }
-    else {
-      return;
-    }
+    return;
   }
 
   public function checkRunning() {
     $queuerunner_pid = (int) \Drupal::state()->get('hydroponics.queurunner_last_pid', 0);
     $lastRunTime = intval(\Drupal::state()->get('hydroponics.heartbeat'));
     $currentTime = intval(\Drupal::time()->getRequestTime());
-    $running_posix = posix_kill($queuerunner_pid, 0);
+    $deltaTime = ($currentTime - $lastRunTime);
 
-    if (!$running_posix || !$queuerunner_pid) {
-     $return = [
-       'running' => FALSE,
-       'message' =>  $this->t('Hydroponics Service Not running, time passed since last seen @time', [
-        '@time' => ($currentTime - $lastRunTime)]),
-        'PID' => $queuerunner_pid
-     ];
-    } else {
+    $heartbeat_max_delta = $this->getHearbeatMaxDelta();
+
+    $running_posix = FALSE;
+    if ($queuerunner_pid > 0) {
```

Comment:
What if $queuerunner_pid is not >0? should we not stop processing and return immediately?

Reply:
Sorry, not clear to me what you mean.